crates/router/src/core/payment_methods/network_tokenization.rs

```diff
@@ -0,0 +1,739 @@
+    cryptogram: Secret<String>,
+    token: CardNumber,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TokenDetails {
+    exp_month: Secret<String>,
+    exp_year: Secret<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TokenResponse {
+    authentication_details: AuthenticationDetails,
+    network: api_enums::CardNetwork,
+    token_details: TokenDetails,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DeleteCardToken {
+    card_reference: String,
+    customer_id: id_type::CustomerId,
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum DeleteNetworkTokenStatus {
+    Success,
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+#[serde(untagged)]
+pub enum DeleteNTResponse {
+    DeleteNetworkTokenResponse(DeleteNetworkTokenResponse),
+    DeleteNetworkTokenErrorResponse(NetworkTokenErrorResponse),
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct NetworkTokenErrorInfo {
+    code: String,
+    developer_message: String,
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct NetworkTokenErrorResponse {
+    error_message: String,
+    error_info: NetworkTokenErrorInfo,
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct DeleteNetworkTokenResponse {
+    status: DeleteNetworkTokenStatus,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CheckTokenStatus {
+    card_reference: String,
+    customer_id: id_type::CustomerId,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum TokenStatus {
+    Active,
+    Inactive,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CheckTokenStatusResponsePayload {
+    token_expiry_month: Secret<String>,
+    token_expiry_year: Secret<String>,
+    token_status: TokenStatus,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CheckTokenStatusResponse {
+    payload: CheckTokenStatusResponsePayload,
+}
+
+pub async fn mk_tokenization_req(
+    state: &routes::SessionState,
+    payload_bytes: &[u8],
+    amount: String,
+    currency: String,
+    customer_id: id_type::CustomerId,
+    tokenization_service: &NetworkTokenizationService,
+) -> CustomResult<(CardNetworkTokenResponsePayload, Option<String>), errors::NetworkTokenizationError>
+{
+    let enc_key = tokenization_service.public_key.peek().clone();
+
+    let key_id = tokenization_service.key_id.clone();
+
+    let jwt = encryption::encrypt_jwe(
+        payload_bytes,
+        enc_key,
+        EncryptionAlgorithm::A128GCM,
+        Some(key_id.as_str()),
+    )
+    .await
+    .change_context(errors::NetworkTokenizationError::SaveNetworkTokenFailed)
+    .attach_printable("Error on jwe encrypt")?;
+
+    let order_data = OrderData {
+        consent_id: uuid::Uuid::new_v4().to_string(),
+        customer_id,
+        amount,
+        currency,
+    };
+
+    let api_payload = ApiPayload {
+        service: "NETWORK_TOKEN".to_string(),
```

Comment:
Avoid hard-coding instead create constants with variable explaining the use case and comments for the created constants